### PR TITLE
Parsed spanids should have type number

### LINKED ISSUE #161: Span ids should be of number type when parsed from headers.

### PR DESCRIPTION
**Pull request by matthewloring**
Fixes #161

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots PTAL.

File: lib/trace-agent.js [hunk 1]
```diff
@@ -202,7 +202,7 @@ TraceAgent.prototype.parseContextFromHeader = function(str) {
   }
   return {
     traceId: matches[1],
-    spanId: matches[2],
+    spanId: Number(matches[2]),
```
Comment (ofrobots) [inline lib/trace-agent.js]: When the argument cannot be converted to a number, `Number(argument)` returns `NaN`. I don't think the Trace API would like seeing `NaN`s here.

File: test/test-trace-agent.js [hunk 2]
```diff
@@ -96,7 +97,7 @@ describe('Trace Agent', function() {
           '123456/667');
         assert(result);
         assert.equal(result.traceId, '123456');
-        assert.equal(result.spanId, '667');
+        assert.equal(result.spanId, 667);
         assert(!result.options);
       });
```
Comment (ofrobots) [inline test/test-trace-agent.js]: Add test(s) to deal with a spanId that cannot be parsed as a number.

File: lib/trace-agent.js (hunk 1 above)
Comment (matthewloring) [inline lib/trace-agent.js]: Fixed.

File: test/test-trace-agent.js (hunk 2 above)
Comment (matthewloring) [inline test/test-trace-agent.js]: I've added the tests to the invalid inputs section below. I think it should be a parse error if we get a non-numeric spanId.

**Comment by ofrobots** [comment]
lgtm

**Pull request merged**